Neos.ContentRepository.Core/Classes/Projection/ContentGraph/ContentGraphWithRuntimeCaches/ContentSubgraphWithRuntimeCaches.php

```diff
@@ -165,27 +165,50 @@ public function findParentNode(NodeAggregateId $childNodeAggregateId): ?Node
 
     public function findNodeByPath(NodePath $path, NodeAggregateId $startingNodeAggregateId): ?Node
     {
-        // TODO implement runtime caches
-        return $this->wrappedContentSubgraph->findNodeByPath($path, $startingNodeAggregateId);
+        // this implementation is copied from the DoctrineAdapter to make it cache-able
+        $startingNode = $this->findNodeById($startingNodeAggregateId);
+
+        return $startingNode
+            ? $this->findNodeByPathFromStartingNode($path, $startingNode)
+            : null;
```

Comment:
AFAIS this change is unrelated!?
Also, the new implementation will be worse for new nodes because we'd look up the starting node multiple times I guess?

A more resilient optimization would be to ask the `NodeByNodeAggregateIdCache` whether it stored the starting node as `nonExistingNodeAggregateIds` (apparently there is no API to retrieve that information yet)..
But I'm not sure if this micro-optimization is worth it and in any case I would strongly suggest not to mix it with this PR